src/util.cc

```diff
@@ -2404,4 +2404,36 @@ bool coredumping_signal_takes_down_entire_vm() {
   return coredumping_signal_vm_behavior > 0;
 }
 
+int parse_tid_from_proc_path(const std::string& pathname,
+                             const std::string& property) {
+  // XXX When rr becomes c++17 - use string view instead. Has better API for this stuff
+  DEBUG_ASSERT(property.back() != '/');
+  DEBUG_ASSERT(pathname.substr(0, 6) == "/proc/");
+  const auto proc_length = 6;
+  const auto prop_begin =
+      pathname.size() - property.size() - (pathname.back() == '/');
+  if (pathname.substr(prop_begin, property.size()) == property) {
+    const auto task = "task/"s;
+    const auto pos = pathname.find(task);
+    if (pos == std::string::npos) {
+      auto tid_length = pathname.find(property) - proc_length;
+      const auto s = pathname.substr(proc_length, tid_length);
+      char* end;
+      const int tid = strtol(s.c_str(), &end, 10);
+      if (!*end) {
```

Comment:
I'm pretty sure strtol sets end to the input pointer on failure[0], not to null. Please check that (e.g. by strtol()ing a letter) and change this code if necessary.

[0] https://c-for-dummies.com/blog/?p=3671

Reply:
You're right, it does. I re-used the previous code that was there. Technically, to check "ok'ness" the check could be 
```cpp
const auto s = pathname.substr(proc_length);
char* end;
const int tid = strtol(s.c_str(), &end, 10); 
if (end == property.c_str()) { ... }
```

Because this way, the "failure" will always be `== property` since the pre-conditions is that the pathname always end with the "property". I'll play around and see what fits best.